delivery_ups_oca/models/ups_request.py

```diff
@@ -158,18 +170,27 @@ def _send_shipping(self, picking):
         )
         self._raise_for_status(status, False)
         res = status["ShipmentResponse"]["ShipmentResults"]
+        PackageResults = res["PackageResults"]
+        if not isinstance(PackageResults, list):
+            PackageResults = [PackageResults]
         return {
             "price": res["ShipmentCharges"]["TotalCharges"],
             "ShipmentIdentificationNumber": res["ShipmentIdentificationNumber"],
-            "GraphicImage": res["PackageResults"]["ShippingLabel"]["GraphicImage"],
+            "PackageResults": PackageResults,
         }
 
     def _quant_package_data_from_order(self, order):
         PackageWeight = 0
         for line in order.order_line.filtered(
             lambda x: x.product_id and x.product_id.weight > 0
         ):
-            PackageWeight += line.product_id.weight * line.product_uom_qty
+            weight_line = line.product_id.weight
+            # Fix if product_dimension_volumetric_weight addon installed
+            if "volumetric_weight" in line.product_id._fields:
```

Comment:
If volumetric weight is admitted by UPS, there should be a hard dependency

Reply:
Volumetric weight is not required by UPS or any carrier, but carriers typically charge based on volumetric weight (because it is often greater than the weight of the package).

With this little code and only for those who have `product_dimension_volumetric_weight` addon installed, that weight will be taken into account instead of the weight of the product. This is very useful, for example, when you want to obtain the price of a quote per API.

Similar to https://github.com/OCA/delivery-carrier/pull/501#discussion_r941648357

Do you think this should be done differently?

Reply:
Hiding a different way to compute data to rate the delivery is indeed an ugly patch and something to avoid. We need to see if UPS/TNT considers this as correct and official, and if so, put the hard dependency and add the explanation in the README.